restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +93,93 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  public static Object fillInDataDefault(DataSchema schema, Object dataWithoutDefault)
+  {
+    try
+    {
+      switch (schema.getType())
+      {
+        case RECORD:
+          return fillInDefaultOnRecord((RecordDataSchema) schema, (DataMap) dataWithoutDefault);
+        case TYPEREF:
+          return fillInDefaultOnTyperef((TyperefDataSchema) schema, dataWithoutDefault);
+        case MAP:
+          return fillInDefaultOnMap((MapDataSchema) schema, (DataMap) dataWithoutDefault);
+        case UNION:
+          return fillInDefaultOnUnion((UnionDataSchema) schema, (DataMap) dataWithoutDefault);
+        case ARRAY:
+          return fillInDefaultOnArray((ArrayDataSchema) schema, (DataList) dataWithoutDefault);
+        default:
+          return dataWithoutDefault;
+      }
+    }
+    catch (CloneNotSupportedException ex)
+    {
+      throw new RestLiServiceException(HttpStatus.S_500_INTERNAL_SERVER_ERROR, ex);
+    }
+  }
+
+  public static DataMap fillInDefaultOnRecord(RecordDataSchema schema, DataMap dataMap) throws CloneNotSupportedException
```

Comment:
change scope to private. not required to be exposed outside. same for other methods